File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Tasks/taskspercentage.rb
module Asposetasksjava
  module TasksPercentage
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'project.mpp')

        # Create a ChildTasksCollector instance
		collector = Rjb::import('com.aspose.tasks.ChildTasksCollector').new

		# Collect all the tasks from RootTask using TaskUtils
		Rjb::import('com.aspose.tasks.TaskUtils').apply(project.getRootTask(), collector, 0)

		tsk = Rjb::import('com.aspose.tasks.Tsk')

		tasks = collector.getTasks()

		# Parse through all the collected tasks
		i = 0
		while i < tasks.size()
			task = tasks.get(i)
			puts task.get(tsk.PERCENT_COMPLETE).toString()
			puts task.get(tsk.PERCENT_WORK_COMPLETE).toString()
			puts task.get(tsk.PHYSICAL_PERCENT_COMPLETE).toString()
			i +=1
		end
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithTasks/TasksPercentage.php
<?php
namespace Aspose\Tasks\WorkingWithTasks;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\ChildTasksCollector as ChildTasksCollector;
use com\aspose\tasks\TaskUtils as TaskUtils;
use com\aspose\tasks\Tsk as Tsk;
class TasksPercentage{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Create a ChildTasksCollector instance
		$collector = new ChildTasksCollector();

        # Collect all the tasks from RootTask using TaskUtils
        $taskUtils=new TaskUtils();
        $taskUtils->apply($project->getRootTask(), $collector, 0);

		$tsk = new Tsk();

		$tasks = $collector->getTasks();

		# Parse through all the collected tasks
		$i = 0;
		while ($i < sizeof($tasks)) {
            $task = $tasks -> get($i);
			print (string)$task -> get($tsk -> PERCENT_COMPLETE).PHP_EOL;// . toString()
			print (string)$task -> get($tsk -> PERCENT_WORK_COMPLETE).PHP_EOL;// . toString()
			print (string)$task -> get($tsk -> PHYSICAL_PERCENT_COMPLETE).PHP_EOL;// . toString()
			$i += 1;
		}

    }


}